pyvista/plotting/lights.py

```diff
@@ -0,0 +1,465 @@
+"""Module containing pyvista implementation of vtkLight."""
+
+from enum import IntEnum
+
+import vtk
+from vtk import vtkLight
+
+import pyvista
+
+class LightType(IntEnum):
+    """An enumeration for the light types."""
+
+    HEADLIGHT = 1
+    CAMERA_LIGHT = 2
+    SCENE_LIGHT = 3
+
+    def __str__(self):
+        """Pretty name for a light type."""
+        return self.name.replace('_', ' ').title()
+
+class Light(vtkLight):
+    """Light class.
+
+    Parameters
+    ----------
+    position : list or tuple, optional
+        The position of the light. The interpretation of the position depends
+        on the type of the light.
+
+    color : list or tuple, optional
+        The color of the light. The ambient, diffuse and specular colors will
+        all be set to this color in creation.
+
+    light_type : string or int, optional
+        The type of the light. If a string, one of ``'headlight'``,
+        ``'camera light'`` or ``'scene light'``. If an int, one of 1, 2 or 3,
+        respectively. The class constants ``Light.HEADLIGHT``, ``Light.CAMERA_LIGHT``
+        and ``Light.SCENE_LIGHT`` are also available, respectively.
+
+        A headlight is attached to the camera, looking at its focal point along
+        the axis of the camera.
+        A camera light also moves with the camera, but it can occupy a general
+        position with respect to it.
+        A scene light is stationary with respect to the scene, as it does not
+        follow the camera. This is the default.
+    """
+
+    # TODO: better/more explanation for ``position``?
+
+    # pull in light type enum values as class constants
+    HEADLIGHT = LightType.HEADLIGHT
+    CAMERA_LIGHT = LightType.CAMERA_LIGHT
+    SCENE_LIGHT = LightType.SCENE_LIGHT
+    # TODO: is this design OK?
+
+    def __init__(self, position=None, color=None, light_type='scene light'):
+        """Initialize the light."""
+        super().__init__()
+
+        if position:
+            self.position = position
+
+        if color:
+            self.ambient_color = color
+            self.diffuse_color = color
+            self.specular_color = color
+            # TODO: make this a single call if that's implemented
```

Comment:
Do they have to be the same color? Could we default to white or black for each, or does that cause issues?

Reply:
They don't _have_ to be, which is why we can set them separately. But if a single "color" is passed the only reasonable thing (to me) is to set every component's color to this.

Our default preserves `vtkLight`'s default, which is black for ambient and white for specular and diffuse. This seems to match a `.SetColor` call with white: that would only set specular and diffuse. If there's reason to leave out ambient unless explicitly requested (as `vtkLight` does it) I can change this too.

As for the TODO comment: I just meant that if `set_color` is changed to set all 3 then I can replace those 3 lines with a single call to `set_color`. We'd still have access to the individual component colour properties.